### ISSUE TITLE
backticks and $(...) aren't highlighted properly

### ISSUE THREAD

**Issue by dimonomid** [zsh-users/zsh-syntax-highlighting#139]
Say, in this command:
`vim $(find -name "something")`
both `vim` and `find` should be highlighted as a command, but only `vim` is.

**Comment by anntzer** [comment]
There's also the process substitution commands, `cmd <(cmd)`, `cmd >(cmd)` and `cmd =(cmd)`.

**Comment by ZevEisenberg** [comment]
Also, commands in strings. I was just bitten by this when I ran the following commit message:

```sh
git commit -m "Add library attributions Settings page that updates whenever `pod install` is run."
```

Of course, the `pod install` was executed, and the output was inserted into my commit message. I wouldn't have run into that problem if zsh-syntax-highlighting had flagged the backtics in the commit message.

**Comment by danielshahaf** [comment]
> Also, commands in double-quoted strings.

Addressed by c3913e0 for the `"$(…)"` syntax.

**Comment by danielshahaf** [comment]
Summary:

_What_ needs highlighting:
- Command substitution ``: `echo foo` ``, ``: $(echo foo)``.
- Command substitution in double quotes: ``"`echo foo`"``, ``"$(echo foo)"`` — the latter case currently highlights the `$` but not the nested command.
- Process substitution `<(foo)`, `=(foo)`, `>(foo)`.

_How_ to highlight:
- TBD.  See #389 for some thoughts on the matter.

**Comment by phy1729** [comment]
> Also, commands in double-quoted strings.

And now addressed for backticks by f16e858. (Although `back-quoted-argument` is not highlighted by default).
